examples/platform/silabs/uart.cpp

```diff
@@ -542,18 +546,18 @@ void uartMainLoop(void * args)
 /**
  * @brief Send Bytes to UART. This blocks the UART task.
  *
- * @param buffer pointer to the buffer containing the data
+ * @param buffer pointer to the UartTxStruct_t containing the data
  * @param nbOfBytes number of bytes to send
  */
-void uartSendBytes(uint8_t * buffer, uint16_t nbOfBytes)
+void uartSendBytes(UartTxStruct_t * bufferStruct, uint16_t nbOfBytes)
 {
 #if SLI_SI91X_MCU_INTERFACE
     // ensuring null termination of buffer
-    if (nbOfBytes != CHIP_SHELL_MAX_LINE_SIZE && buffer[nbOfBytes - 1] != '\0')
+    if (nbOfBytes < sizeof(bufferStruct->data) && bufferStruct->data[nbOfBytes - 1] != '\0')
```

Comment:
sizeof(bufferStruct->data) this would give you 4 as size of the pointer no ?
if for some reason not caught before, nbOfBytes is > than bufferStruct->data we aren't ensuring that bufferStruct->data is null terminated and we are still passing it to Board_UARTPutSTR. That will cause out of memory access because I believe Board_UARTPutSTR depends on the null char

Reply:
size data is a defined sized array : data[UART_TX_MAX_BUF_LEN], this will return the total size of the entire array.